### PR TITLE
Fix issue #6511: Adaptive Icons

### PR DESCRIPTION
**Pull request by yvan-sraka**
### First time contributor checklist
- [x] I have read [how to contribute](https://github.com/WhisperSystems/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [x] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist
- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution on these devices:
 * Nexus 5, Android O
- [x] My contribution is fully baked and ready to be merged as is
- [x] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in the commit message of my first commit

----------

### Description

Introduces the new [Adaptive Icons](https://developer.android.com/preview/features/adaptive-icons.html) feature of Android O!

### CODE REVIEW

**Comment by moxie0** [comment]
Thanks, I don't have any Android O setup yet, can you provide screenshots for how this looks?

**Comment by mueller-ma** [comment]
I'm not a expert, but this seems wrong to me. It is just the current icon (with background) and a black adaptive background. The foreground icon should only be the speech bubble without any blue. The background should be the Signal blue.

And providing two icons (square and round) is not needed and exactly what adaptive icons are not. You just provide ONE foreground and ONE background and every launcher can show it in square, round, etc.

https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive.html

**Comment by yvan-sraka** [comment]
You're right, as I mentioned in issue #6511, I do not have graphic designer skills, so I've done my work with [the only flat file of the application icon that I had at my disposal](
https://github.com/WhisperSystems/Signal-Android/blob/master/artwork/logo-512.png). I would really appreciate any improvements that could be made to this PR. But as a first implementation, I also want to provide a standard icon that follows adaptive icon guidelines to provide both design consistency and a retro-compatibility (because the adaptive icon requires minSDK 26)!

**Comment by moxie0** [comment]
@yvan-sraka Sorry I'm not sure that I understand what it is you're trying to accomplish with this PR. Could you try restating (perhaps with screenshots) what this PR does and why it makes sense if it's not following the icon design guidelines? Thanks!

**Comment by mueller-ma** [comment]
Well, we can just add a adaptive icon and leave the others.

**Comment by mueller-ma** [comment]
Use https://github.com/WhisperSystems/Signal-Android/blob/master/res/drawable-xxxhdpi/icon_transparent.png as foreground and #359AEB as background

**Comment by yvan-sraka** [comment]
Thanks @mueller-ma, I updated my previous commit following your advice!

**Comment by mueller-ma** [comment]
AFAIK you only need to create res/drawable-anydpi/ic_launcher.xml
````
<?xml version="1.0" encoding="utf-8"?>
<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="#359AEB"/>
    <foreground android:drawable="@drawable/icon_transparent"/>
</adaptive-icon>
````

**Comment by Stanzilla** [comment]
Are you sure you can change the targetSdkVersion without also adopting other Oreo features like notification channels? I remember them saying that if you target the new SDK you will have to support them or notifications won't work. And in general, I think a few other things will have to be changed if you update the targetSdkVersion.

**Comment by rainerzufall** [comment]
An sdk change like in this PR was just recently rejected https://github.com/WhisperSystems/Signal-Android/pull/6901 Moxie said:
> Many  many things would have to change in order to target even API 23.

File: build.gradle [hunk 1]
```diff
@@ -207,7 +207,7 @@ android {
         versionName "4.9.9"
 
         minSdkVersion 9
-        targetSdkVersion 22
+        targetSdkVersion 26
```
Comment (moxie0) [inline build.gradle]: Please revert this.

Comment (mueller-ma) [reply build.gradle]: Adaptive icons work only with 26+

**Comment by yvan-sraka** [comment]
@Stanzilla, @rainerzufall, I revert the targetSdkVersion changes on my PR to give Signal an update, compatible with current Android versions, that allows the Signal icon to not break the design consistency on new devices!

**Comment by fercarcedo** [comment]
Actually, in order to provide adaptive icons, it's not necessary to set targetSdkVersion to 26, you just need to upgrade the compileSdkVersion from the current (25) to 26 and that's it, you can keep target on 22. For more info:

https://medium.com/@ianhlake/vectordrawable-adaptive-icons-3fed3d3205b5

**Comment by moxie0** [comment]
@yvan-sraka Have you tested this? I'm doubtful that it even compiles.

**Comment by mueller-ma** [comment]
Can you look at this commit https://github.com/mueller-ma/Signal-Android/commit/f45cce554d0254b439012f1a39b9470870439b9a ? This should be enough. Can someone test it? I dont have a Android 8 device.

**Comment by Stanzilla** [comment]
@mueller-ma are you going to make a PR?

**Comment by yvan-sraka** [comment]
@mueller-ma I followed your advices, and reduce the PR to your changes suggestion, I let one of my friends @MattiaPrimavera confirm that it compiles and works well!

**Pull request closed without merging**